include/ibamr/INSVCStaggeredConservativeConvectiveOperator.h

```diff
@@ -196,6 +197,29 @@ class INSVCStaggeredConservativeConvectiveOperator : public ConvectiveOperator
      */
     void setMassDensitySourceTerm(const SAMRAI::tbox::Pointer<IBTK::CartGridFunction> S_fcn);
 
+    /*!
+     * \brief Set the patch data indices corresponding to the velocity at the previous time step
+     * to be used when computing the density update
+     *
+     * \note This velocities will be used to compute an approximation to velocities required for SSPRK updates
+     * V_old_idx = n-1, V_current_idx = n, V_new_idx = n+1,k (after an INS cycle)
+     * If V_old_idx or V_new_idx are not set, then they will degenerate to V_current automatically, for the very first
+     * simulation time step and cases where an INS cycle has not been executed, respectively.
+     */
+    void setFluidVelocityPatchDataIndices(int V_old_idx, int V_current_idx, int V_new_idx);
+
+    /*!
+     * \brief Set the cycle number currently being executed by the INS integrator. This will determine the rho advection
+     * velocity.
+     */
+    void setCycleNumber(int cycle_num);
+
+    /*!
+     * \brief Set the previous time step value between times n - 1 (old) and n (current). This is used during velocity
+     * extrapolation.
+     */
+    void setPreviousTimeStepSize(double dt_prev);
```

Comment:
Velocity extrapolation should not be done in this operator.

Reply:
We need velocity extrapolation for the RK scheme for density update. There is a comment in this routine which says:

// Compute the convective derivative with the penultimate density and velocity

We compute div. (rho u u) from an intermediate/penultimate stage  of mass equation, i.e, intermediate value is more important than the actual density update completion. We need this for stability. Hence, we are doing it simultaneously in the same routine.

Reply:
I guess my only question is whether these values could be accessed through a pointer to the parent `HierarchyIntegrator`, instead of having an interface to set the values. If you can get them from the parent integrator, it seems easier just to get as needed, rather than requiring explicit passing from the parent class.

Reply:
Hmm, the only way I see is via friendship. This has to be done down the lane as friendship of base class (future TimeIntegrator base class) won't make the derived class a friend of XXXHierarchyIntegrator. At that point, one may just specialize the XXXTimeIntegrator to obtain patch indices of XXXHierarchyIntegrator.